source/smeagle.cpp

```diff
@@ -21,29 +21,26 @@ using namespace smeagle;
 Smeagle::Smeagle(std::string _library) : library(std::move(_library)) {}
 
 // Parse the library with smeagle
-int Smeagle::parse(FormatCode fmt) {
+smeagle::Corpus Smeagle::parse() {
   // We are going to read functions and symbols
   Symtab *obj = NULL;
   std::vector<Symbol *> symbols;
   std::vector<Function *> funcs;
 
   // Read the library into the Symtab object, cut out early if there's error
   if (not Symtab::openFile(obj, library)) {
-    std::cout << "There was a problem reading " << library << "\n";
-    return 1;
+    throw "There was a problem reading the library..";
```

Comment:
It's legal, but strongly discouraged to throw primitive types. I would throw a runtime_error here.

    throw std::runtime_error{"There was a problem reading from '"+library+"'"};
    
I would also `#include <stdexcept>` just for good measure- I think it comes with `<string>`, but I can't remember.

Reply:
ahh didn't know about that, thank you!

Reply:
Okay all set https://github.com/buildsi/Smeagle/pull/13/commits/a74a799ad9d9eeca0d484305c247871662437056

Reply:
LGTM!

Reply:
woot! I'll merge when testing is finished and get started updating the other PR.